app/actions/user.js

```diff
@@ -1,8 +1,79 @@
+import{
+  AsyncStorage
+}from 'react-native';
 import * as types from './types';
+const BreakoutApi= require("breakout-api-client");
 
 export function setUser(user){
   return{
     type: types.SET_USER,
     payload:user
   }
 }
+
+export function setUsername(username){
+  return {
+    type: types.SET_USERNAME,
+    payload:username
+  }
+}
+
+export function setLoggingIn(logging){
+  return {
+    type: types.LOGGING_IN,
+    payload:logging
+  }
+}
+
+export function login(username,password){
+  return (dispatch,getState) => {
+    console.log(username+" "+password);
+    let curState = getState()
+    let loggingIn = curState.isLoggingIn;
+    console.log(BreakoutApi);
+    dispatch(setLoggingIn(true));
+    sendLogin(username,password).then((resp)=>{
+      if(resp.success === 1){
+        console.log("success");
+        saveUserData(username,password).then(()=>{
```

Comment:
Why are we saving username and password instead of the received tokens? Are we planning on reusing the same BreakoutApi object over the whole app or are we initializing it every time we use it?
If so, it can be initialized with `api.setAccessToken(accessToken)` in order to save the additional login call to the backend on each request

Reply:
Okay, in this case I'll add the tokens to the saved data